samcli/lib/sync/flows/image_function_sync_flow.py

```diff
@@ -82,7 +81,14 @@ def gather_resources(self) -> None:
         )
         self._image_name = builder.build().artifacts.get(self._function_identifier)
         if self._image_name:
-            self._local_sha = str_checksum(self._image_name)
+            self._local_sha = self._get_local_image_id(self._image_name)
+
+    def _get_local_image_id(self, image: str) -> Optional[str]:
+        """Returns the local hash of the image"""
+        docker_img = self._get_docker_client().images.get(image)
```

Comment:
Note for other reviewers: we do not need exception handling here because a failed execution of the sync flow is wrapped by an exception handler that will automatically trigger infra sync, and will not break --watch process by design